docs/deploys.md

```diff
@@ -0,0 +1,120 @@
+# Halyard Deploys
+
+## Purpose
+
+Explain how Halyard handles deploying Spinnaker.
+
+## Terminology
+
+* **Deployment Type**: How Spinnaker is divided among all the instances it is
+ running on. An example is having each Spinnaker service run a separate
+ instance. See [Deployment Types](#deployment-types) section for more
+ examples.
+* **Service Dependencies**: Dependencies of Spinnaker that need to be run as
+ stand-alone services. An example is Redis, a non-example is Packer.
+* **Deployment Environment**: The infrastructure Spinnaker is running on,
+ Spinnaker's Deployment Type, and Spinnaker's service dependencies.
+
+## Deployment Types
+
+* **LocalhostDebian**: This is every Spinnaker service installed as a Debian
+ package on the node Halyard is running on.
+* **LocalhostGit**: This is every Spinnaker service installed from Git on the 
+ node Halyard is running on. This is intended for use by Developers.
+* **Flotilla**: This is every Spinnaker service installed remotely on a
+ specified Cloud Provider as separate, independently scalable instance. There
+ is no customization or breakout of specialized profiles for this deployment.
+
+Every deployment type must address a few issues:
+
+1. **Configuration**: How Spinnaker recieves the config it needs to run.
+2. **Service Dependency Setup**: What services Spinnaker must install to get
+ this deployment to run.
+3. **Deployment & Updates**: How the Spinnaker services are deployed and
+ updated in this environment.
+
+### LocalhostDebian Deployments
+
+> Timeline: Already included in Halyard.
+
+Halyard handles this deployment type by generating an install script very
+similar to `InstallSpinnaker.sh` in spinnaker/spinnaker repository. The CLI
+receieves this script, and attempts to run it locally. 
+
+1. **Configuration**: The script generated by Halyard has commands to copy all
+ configuration from the necesssary Spinnaker config already generated by 
+ Halyard into `/opt/spinnaker/config`, `/opt/spinnaker-monitoring` and 
+ `/etc/apache2`. Any required files are already in the correct location.
+2. **Service Dependency Setup**: The only service dependency is Redis, and it
+ is installed locally.
+3. **Deployment & Updates**: All spinnaker packages are pinned by writing
+ `apt` preferences in `/etc/apt/preferences.d/pin-spin-{artifact}` for each
+ Spinnaker artifact. During the initial install and subsequent updates, this
+ file is always updated before attempting to grab new packages via `apt`.
+
+### LocalhostGit Deployments
+
+> Timeline: Mid-to-late April
+
+Halyard handles this deployment by installing all Spinnaker service git repos
+in a user-configurable directory, with a user-configurable `upstream` and
+`origin`.  There will be a special version called `HEAD` that supersedes any 
+configuration in our profile registry that will take all services at git `HEAD` 
+to use for config generation.
+
+### Flotilla Deployments
+
+> Timeline: Varies by cloud provider. See below.
```

Comment:
The tone of this section has changed from your original purpose of explaining how it is done to a discussion about the issues